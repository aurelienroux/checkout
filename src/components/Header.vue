<template>
  <header class="header">
    <div class="header__top">
      <span>allbirds</span>
      <a href="https://www.allbirds.ca/" target="_blank" rel="noopener noreferrer">visit allbirds.ca --></a>
    </div>
    <h1>Jenny's outfit</h1>
    <h2>Dress like your idole with this super bundle and be the star you always wanted to be!</h2>
  </header>
</template>

<style lang="scss" scoped>
.header {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 1rem;

  &__top {
    align-items: center;
    display: flex;
    flex: 1;
    justify-content: space-between;

    span {
      font-size: 2rem;
    }
  }

  h2 {
    max-width: 40rem;
  }
}
</style>
