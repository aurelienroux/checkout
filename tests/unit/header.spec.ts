import { shallowMount } from '@vue/test-utils'
import Header from '@/components/Header.vue'

const wrapper = shallowMount(Header)

describe('Header component', () => {
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h1').exists()).toBe(true)
    expect(wrapper.find('h2').exists()).toBe(true)
  })
})
