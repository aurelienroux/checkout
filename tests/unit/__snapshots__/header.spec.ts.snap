// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders without crashing 1`] = `
<header class="header">
  <div class="header__top"><span>allbirds</span> <a href="https://www.allbirds.ca/" target="_blank" rel="noopener noreferrer">visit allbirds.ca --&gt;</a></div>
  <h1>Jenny's outfit</h1>
  <h2>Dress like your idole with this super bundle and be the star you always wanted to be!</h2>
</header>
`;
